tests/lib/rules/semi-spacing.js

```diff
@@ -206,7 +253,74 @@ ruleTester.run("semi-spacing", rule, {
             options: [{ before: false, after: true }],
             parserOptions: { ecmaVersion: 6, sourceType: "module" },
             errors: [
-                { messageId: "unexpectedWhitespaceBefore", type: "ExportDefaultDeclaration", line: 1, column: 20 }
+                { messageId: "unexpectedWhitespaceBefore", type: "ExportDefaultDeclaration", line: 1, column: 19 }
+            ]
+        },
+        {
+            code: "while(foo) {continue   ;}",
+            output: "while(foo) {continue;}",
+            options: [{ before: false, after: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "unexpectedWhitespaceBefore",
+                    type: "ContinueStatement",
+                    line: 1,
+                    column: 21,
+                    endColumn: 24
+                }
+            ]
+        },
+        {
+            code: "if(foo) {throw new Error()   ;  }",
+            output: "if(foo) {throw new Error();  }",
+            options: [{ before: false, after: false }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "unexpectedWhitespaceBefore",
+                    type: "ThrowStatement",
+                    line: 1,
+                    column: 27,
+                    endColumn: 30
+                }
+            ]
+        },
+        {
+            code: "for(a ; ; );",
+            output: "for(a;; );",
+            options: [{ before: false, after: false }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{
+                line: 1,
+                column: 6,
+                type: "ForStatement",
+                messageId: "unexpectedWhitespaceBefore",
+                endLine: 1,
+                endColumn: 7
+            },
```

Comment:
Can we group `line`, `column`, `endLine`, `endColumn` lines, for consistency.

(the same applies to some other tests)